class-components/src/components/SelectedFlyout/SelectedFlyout.tsx

```diff
@@ -1,54 +1,68 @@
+'use client';
+
 import { useAppDispatch, useAppSelector } from '../../hooks/reduxHook';
 import type { Character } from '../../types';
 import { clearSelected } from '../../providers/redux/selectedItemsSlice';
-import './style.css';
+import styles from './SelectedFlyout.module.css';
+import { useTranslations } from 'next-intl';
+import { useEffect, useActionState, useMemo } from 'react';
+import { buildCsvAction } from '../../app/actions/exportCsv';
 
 interface SelectedFlyoutProps {
   items: Character[];
 }
 
 export const SelectedFlyout = ({ items }: SelectedFlyoutProps) => {
+  const t = useTranslations('Select');
   const dispatch = useAppDispatch();
   const selectedIds = useAppSelector(
     (state) => state.selectedItems.selectedIds
   );
-  if (selectedIds.length === 0) return null;
 
-  const selectedData = items.filter((char) => selectedIds.includes(char.id));
+  const selectedData = useMemo(
+    () => items.filter((c) => selectedIds.includes(c.id)),
+    [items, selectedIds]
+  );
```

Comment:
если не хочешь имя прописывать то лучше так
```suggestion
  const selectedData = useMemo(
    () => items.filter(({ id }) => selectedIds.includes(id)),
    [items, selectedIds]
  );
```